documentation/src/main/asciidoc/topics/con_performance_indexing_querying.adoc

```diff
@@ -0,0 +1,39 @@
+[id='performance-indexing-querying_{context}']
+= Indexing and querying caches
+
+{brandname} lets you query across caches so you can analyze and filter your data, gaining real-time insights.
+As an example, consider an online game where players compete against each other in some way to score points.
+If you wanted to implement a leaderboard with the top ten players at any one time, you could create a query to find out which players have the most points at any one time and limit the result to a maximum of ten as follows:
+
+[source,java,options="nowrap",subs=attributes+]
+----
+include::code_examples/QueryPlayerScore.java[]
+----
+
+The preceding example illustrates the benefit of using queries because it lets you find ten entries that match a criteria out of potentially millions of cache entries.
+
+In terms of performance impact, though, you should consider the tradeoffs that come with indexing operations versus query operations.
+Configuring {brandname} to index caches results in much faster queries.
+You can query millions of cache entries in less than a millisecond if you use indexes.
+Without indexes, though, queries are at least ten times slower.
+
+Indexing your caches affects write operations.
+Writes are not an order of magnitude slower with indexing but there is a measurable loss of performance.
+However, with some careful planning and a good understanding of what you want to index, you can avoid the worst effects.
+
+The most effective approach is to configure {brandname} to index only the fields that you need.
+Whether you store Plain Old Java Objects (POJOs) or use Protobuf schema, the more fields that you annotate, the longer it takes {brandname} to build the index.
+If you have a POJO with five fields but you only need to query two of those fields, do not configure {brandname} to index the three fields you don't need.
+Likewise, using analyzers can add to the workload for building indexes.
+If an analyzer is overly complicated, for instance one that splits and modifies fields, it can add even more overhead to indexing operations.
+
+{brandname} gives you several options to tune indexing operations.
+For instance {brandname} stores indexes differently to data, saving indexes to disk instead of memory.
+After it builds an index, {brandname} keeps it synchronized with the cache using an index writer.
+If you enable indexing and then observe slower writes, and think indexing causes the loss of performance, you can keep indexes in a memory buffer for longer periods of time before writing to disk.
```

Comment:
A question for @gustavonalle @anistor 
Is there a risk to loose index entries if the node keeps a lot in memory buffer and crash before it is written to the persistence?